### PR TITLE
grpc server [1]: update kvproto, remove kv/coprocessor from tcp server.

### PR DESCRIPTION
**Pull request by disksing**
- Update kvproto.
- Remove usage of `Message`, which may be RaftMessage, KvMessage or CoprocessorMessage.
- Remove kv/coprocessor from TCP server, now it only support RaftMessage.

Note: Now this branch can build and test, but can't work with tidb.

The plan:
1. [x] update kvproto.
2. [x] remove kv/coprocessor from TCP server.
3. [ ] impl  grpc server.
4. [ ] impl raft grpc client.
5. [ ] impl grpc snapshot.
6. [ ] replace TCP server with grpc servers.
7. [ ] add tests.

### CODE REVIEW

**Comment by siddontang** [comment]
why not support gRPC for RaftMessage directly?

**Comment by disksing** [comment]
@siddontang 
I want to split it into small PRs, the very first PR mainly focus on updating kvproto and while ensuring that it can be compiled.

File: src/server/coprocessor/endpoint.rs [hunk 1]
```diff
@@ -997,7 +993,7 @@ impl<'a> SelectContext<'a> {
             } else {
                 range.get_start().to_vec()
             };
-            let upper_bound = if !self.core.desc_scan && range.has_end() {
+            let upper_bound = if !self.core.desc_scan && !range.get_end().is_empty() {
```
Comment (siddontang) [inline src/server/coprocessor/endpoint.rs]: can the end key be empty?

Comment (disksing) [inline src/server/coprocessor/endpoint.rs]: I suppose it won't be empty in normal case.
proto3 doesn't have `has_xx()` so I use `is_empty()` as an equivalent replacement.

**Comment by siddontang** [comment]
LGTM

**Comment by hhkbp2** [comment]
LGTM
CI fails

**Comment by disksing** [comment]
@hhkbp2 phase3 won't pass test anyway, I'm going to merge it and fix CI later in branch grpc.

**Pull request merged**